common/src/main/java/io/netty/util/concurrent/FastThreadLocal.java

```diff
@@ -147,19 +148,27 @@ public final V get() {
 
     private void registerCleaner(final InternalThreadLocalMap threadLocalMap) {
         Thread current = Thread.currentThread();
-        if (!FastThreadLocalThread.willCleanupFastThreadLocals(current)) {
-            // We will need to ensure we will trigger remove(InternalThreadLocalMap) so everything will be released
-            // and FastThreadLocal.onRemoval(...) will be called.
-            ObjectCleaner.register(current, new Runnable() {
-                @Override
-                public void run() {
-                    remove(threadLocalMap);
-
-                    // It's fine to not call InternalThreadLocalMap.remove() here as this will only be triggered once
-                    // the Thread is collected by GC. In this case the ThreadLocal will be gone away already.
-                }
-            });
+        if (FastThreadLocalThread.willCleanupFastThreadLocals(current)) {
+            return;
+        }
+        Object v = threadLocalMap.indexedVariable(addToCleanerFlagIndex);
+        if (v != InternalThreadLocalMap.UNSET) {
+            return;
```

Comment:
@kojilin so the InternalThreadLocalMap is always the same instance?

Reply:
ah...I misunderstood about this, I need to add flag for each index.

Reply:
Looks like I need to remove for each index.